### PR TITLE
Adding pre-commit job to Zuul 

### PR DESCRIPTION
**Pull request by shreyaspapi**
#13 Adding pre-commit to zuul as did in other repos.

### CODE REVIEW

File: .zuul.yaml [hunk 1]
```diff
@@ -4,6 +4,20 @@
     check:
       jobs:
         - noop
+        - dashboard-pre-commit
```
Comment (jpopelka) [inline .zuul.yaml]: Shouldn't it replace the `noop` job?
```suggestion
        - dashboard-pre-commit
```

File: files/zuul-pre-commit.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+- name: Run pre-commit
+  hosts: all
+  tasks:
+    - include_tasks: tasks/zuul-project-setup.yaml
+    - name: Install git
+      dnf:
+        name:
+          - git-core
+          - python36
```
Comment (jpopelka) [inline files/zuul-pre-commit.yaml]: I'm just wondering why we need `python36`.
I know you only copied it from some other repo, but I'd just try to go without it.

Comment (shreyaspapi) [reply files/zuul-pre-commit.yaml]: ohh yes, I will remove it

File: .zuul.yaml [hunk 3]
```diff
@@ -3,7 +3,20 @@
     name: packit-service/dashboard
     check:
       jobs:
-        - noop
+        - dashboard-pre-commit
     gate:
       jobs:
         - noop
+        - dashboard-pre-commit
```
Comment (jpopelka) [inline .zuul.yaml]: There's one more :-)

**Comment by lachmanfrantisek** [comment]
> Please squash the commits into one and I'll merge it.

Don't we want to wait for zuul and actually try that?

**Comment by shreyaspapi** [comment]
Please don't merge until I do the necessary changes to https://softwarefactory-project.io/r/config/

**Comment by shreyaspapi** [comment]
@lachmanfrantisek I have added the repo for zuul to recognize. Let's wait for it to get merged.

**Comment by shreyaspapi** [comment]
Referencing the review - https://softwarefactory-project.io/r/#/c/17738/

**Comment by lachmanfrantisek** [comment]
recheck

**Comment by lachmanfrantisek** [comment]
:tada: Looks like zuul is connected! :tada:

Have you found some problems/notes during the setup? If yes, please update the guide in https://github.com/packit-service/research repo and send a PR. Thanks!

**Comment by shreyaspapi** [comment]
It is supposed to run the test noop job right?

**Comment by lachmanfrantisek** [comment]
> It is supposed to run the test noop job right?

It already ran it in the other PR. We can probably go on with the real jobs.

**Comment by shreyaspapi** [comment]
Ohh great! so I guess we can merge this PR then. You want me to add any other jobs?

**Comment by lachmanfrantisek** [comment]
> Ohh great! so I guess we can merge this PR then. You want me to add any other jobs?

We need to fix it at first. Look at the result -- it fails for some reason.

**Comment by shreyaspapi** [comment]
It's working now, I will rebase. 

I am still confused about why it's not able to find python3.6 interpreter, its already supposed to be installed right?

**Comment by shreyaspapi** [comment]
> The latest build succeeded, so is the python36 problem solved and can we merge?

Yes, It did solve the problem 😄

**Comment by jpopelka** [comment]
Ok, great. Let's merge then and see this in action. Thank you!

**Pull request merged**